// aws-amplify-angular/dist/src/components/authenticator/confirm-sign-up-component/confirm-sign-up.component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
var template = "\n<div class=\"amplify-container\" *ngIf=\"_show\">\n<div class=\"amplify-form-container\">\n  <div class=\"amplify-form-body\">\n    <div class=\"amplify-form-header\">Confirm Sign up</div>\n\n    <div class=\"amplify-form-row\">\n      <label class=\"amplify-input-label\" for=\"amplifyUsername\"> Username *</label>\n      <input\n        #amplifyUsername\n        class=\"amplify-form-input\"\n        type=\"text\"\n        disabled\n        placeholder=\"Username\"\n        [value]=\"username\"\n      />\n    </div>\n    <div class=\"amplify-form-row\">\n      <label class=\"amplify-input-label\" for=\"code\"> Confirmation Code *</label>\n      <input #code\n        (change)=\"setCode(code.value)\"\n        (keyup)=\"setCode(code.value)\"\n        (keyup.enter)=\"onConfirm()\"\n        class=\"amplify-form-input\"\n        type=\"text\"\n        placeholder=\"Enter your Code\"\n      />\n      <span class=\"amplify-form-action\">Lost your code?\n        <a class=\"amplify-form-link\"\n            (click)=\"onResend()\"\n          >Resend Code</a></span>    \n    </div>\n    \n    <div class=\"amplify-form-actions\">\n      <div class=\"amplify-form-cell-left\">\n        <div class=\"amplify-form-actions-left\">\n          <a class=\"amplify-form-link\" (click)=\"onSignIn()\">Back to Sign in</a>\n        </div>\n      </div>\n\n      <div class=\"amplify-form-cell-right\">\n        <button class=\"amplify-form-button\"\n          (click)=\"onConfirm()\">Confirm</button>\n      </div>\n    </div>\n\n  </div>\n\n</div>\n\n<div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n  <div class=\"amplify-alert-body\">\n    <span class=\"amplify-alert-icon\">&#9888;</span>\n    <div class=\"amplify-alert-message\">{{ errorMessage }}</div>\n    <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n  </div>\n</div>\n\n</div>\n";
var ConfirmSignUpComponentCore = /** @class */ (function () {
    function ConfirmSignUpComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(ConfirmSignUpComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'confirmSignUp';
            this.username = data.authState.user ? data.authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ConfirmSignUpComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'confirmSignUp';
            this.username = authState.user ? authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    ConfirmSignUpComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    ConfirmSignUpComponentCore.prototype.setCode = function (code) {
        this.code = code;
    };
    ConfirmSignUpComponentCore.prototype.onConfirm = function () {
        var _this = this;
        this.amplifyService.auth()
            .confirmSignUp(this.username, this.code)
            .then(function () { return console.log('confirm success'); })
            .catch(function (err) { return _this._setError(err); });
    };
    ConfirmSignUpComponentCore.prototype.onResend = function () {
        var _this = this;
        this.amplifyService.auth().resendSignUp(this.username)
            .then(function () { return console.log('code resent'); })
            .catch(function (err) { return _this._setError(err); });
    };
    ConfirmSignUpComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    ConfirmSignUpComponentCore.prototype.onAlertClose = function () {
        this._setError(null);
    };
    ConfirmSignUpComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    ConfirmSignUpComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-confirm-sign-up-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    ConfirmSignUpComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    ConfirmSignUpComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return ConfirmSignUpComponentCore;
}());
export { ConfirmSignUpComponentCore };
