// aws-amplify-angular/dist/src/components/authenticator/confirm-sign-in-component/confirm-sign-in-component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-container\">\n    <div class=\"amplify-form-body\">\n    <div class=\"amplify-form-header\">Confirm Sign in</div>\n\n      <div class=\"amplify-form-row\">\n        <label class=\"amplify-input-label\" for=\"code\"> Confirmation Code *</label>\n        <input #code\n          (change)=\"setCode(code.value)\"\n          (keyup)=\"setCode(code.value)\"\n          (keyup.enter)=\"onConfirm()\"\n          class=\"amplify-form-input\"\n          type=\"text\"\n          placeholder=\"Enter your Code\"\n        />\n      </div>\n\n      <div class=\"amplify-form-actions\">\n        <div class=\"amplify-form-cell-left\">\n          <div class=\"amplify-form-actions-left\">\n            <a class=\"amplify-form-link\" (click)=\"onSignIn()\">Back to Sign in</a>\n          </div>\n        </div>\n\n        <div class=\"amplify-form-cell-right\">\n          <button class=\"amplify-form-button\"\n            (click)=\"onConfirm()\">Confirm</button>\n        </div>\n      </div>\n\n      </div>\n  </div>\n\n\n<div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n  <div class=\"amplify-alert-body\">\n    <span class=\"amplify-alert-icon\">&#9888;</span>\n    <div class=\"amplify-alert-message\">{{ errorMessage }}</div>\n    <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n  </div>\n</div>\n</div>\n";
var ConfirmSignInComponentCore = /** @class */ (function () {
    function ConfirmSignInComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(ConfirmSignInComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'confirmSignIn';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ConfirmSignInComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'confirmSignIn';
        },
        enumerable: true,
        configurable: true
    });
    ConfirmSignInComponentCore.prototype.setCode = function (code) {
        this.code = code;
    };
    ConfirmSignInComponentCore.prototype.onConfirm = function () {
        var _this = this;
        var user = this._authState.user;
        var challengeName = user.challengeName;
        var mfaType = challengeName === 'SOFTWARE_TOKEN_MFA' ? challengeName : null;
        this.amplifyService.auth()
            .confirmSignIn(user, this.code, mfaType)
            .then(function () {
            _this.amplifyService.setAuthState({ state: 'signedIn', user: user });
        })
            .catch(function (err) { return _this._setError(err); });
    };
    ConfirmSignInComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    ConfirmSignInComponentCore.prototype.onAlertClose = function () {
        this._setError(null);
    };
    ConfirmSignInComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    ConfirmSignInComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-confirm-sign-in-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    ConfirmSignInComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    ConfirmSignInComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return ConfirmSignInComponentCore;
}());
export { ConfirmSignInComponentCore };
