// aws-amplify-angular/dist/src/components/authenticator/sign-up-component/sign-up.component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers';
import { countrylist } from '../../../assets/countries';
import defaultSignUpFieldAssets from '../../../assets/default-sign-up-fields';
var template = "\n<div class=\"amplify-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-container\">\n    <div class=\"amplify-form-body\">\n      <div class=\"amplify-form-header\">{{this.header}}</div>\n\n      <div class=\"amplify-form-row\" *ngFor=\"let field of signUpFields\">\n        <div *ngIf=\"field.key !== 'phone_number'\">\n          <label class=\"amplify-input-label\">\n            {{field.label}} \n            <span *ngIf=\"field.required\">*</span>\n          </label>\n          <input #{{field.key}}\n            class=\"amplify-form-input\"\n            [ngClass]=\"{'amplify-input-invalid ': field.invalid}\"\n            type={{field.type}}\n            placeholder={{field.label}}\n            [(ngModel)]=\"user[field.key]\" name=\"field.key\" />\n        </div>\n        <div *ngIf=\"field.key === 'phone_number'\">\n          <label class=\"amplify-input-label\">\n            {{field.label}} \n            <span *ngIf=\"field.required\">*</span>\n          </label>\n          \n          <div class=\"amplify-input-group\">\n            <div class=\"amplify-input-group-item\">\n              <select #countryCode\n                name=\"countryCode\" \n                [ngClass]=\"{'amplify-input-invalid ': field.invalid}\"\n                class=\"amplify-select-phone-country\" \n                [(ngModel)]=\"country_code\">\n                <option *ngFor=\"let country of countries\"  \n                  value={{country.value}}>{{country.label}} \n                </option>\n              </select>\n            </div>\n            <div class=\"amplify-input-group-item\">\n              <input \n                class=\"amplify-form-input\"\n                placeholder={{field.label}}\n                [ngClass]=\"{'amplify-input-invalid ': field.invalid}\"\n                [(ngModel)]=\"local_phone_number\"\n                name=\"local_phone_number\"\n                type={{field.type}}\n              />\n            </div>\n          </div>\n        </div>\n      </div>\n      <div class=\"amplify-form-actions\">\n        \n        <div class=\"amplify-form-cell-left\">\n          <div class=\"amplify-form-signup\">\n            Have an account? <a class=\"amplify-form-link\" (click)=\"onSignIn()\">Sign in</a>\n          </div>\n        </div>\n\n        <div class=\"amplify-form-cell-right\">\n          <button class=\"amplify-form-button\"\n          (click)=\"onSignUp()\"\n          >Sign Up</button>\n        </div>\n\n      </div>\n\n    </div>\n\n  </div>\n\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ errorMessage }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n\n</div>\n";
var SignUpField = /** @class */ (function () {
    function SignUpField() {
    }
    return SignUpField;
}());
export { SignUpField };
var SignUpComponentCore = /** @class */ (function () {
    function SignUpComponentCore(amplifyService) {
        this.user = {};
        this.country_code = '1';
        this.header = 'Create a new account';
        this.defaultSignUpFields = defaultSignUpFieldAssets;
        this.signUpFields = this.defaultSignUpFields;
        this.hiddenFields = [];
        this.countries = countrylist;
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(SignUpComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'signUp';
            if (data.signUpConfig) {
                this._signUpConfig = data.signUpConfig;
                if (this._signUpConfig.defaultCountryCode) {
                    this.country_code = this._signUpConfig.defaultCountryCode;
                }
                if (this._signUpConfig.signUpFields) {
                    this.signUpFields = this._signUpConfig.signUpFields;
                }
                if (this._signUpConfig.header) {
                    this.header = this._signUpConfig.header;
                }
                if (this._signUpConfig.hiddenDefaults) {
                    this.hiddenFields = this._signUpConfig.hiddenDefaults;
                }
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SignUpComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'signUp';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(SignUpComponentCore.prototype, "signUpConfig", {
        set: function (signUpConfig) {
            if (signUpConfig) {
                this._signUpConfig = signUpConfig;
                if (this._signUpConfig.defaultCountryCode) {
                    this.country_code = this._signUpConfig.defaultCountryCode;
                }
                if (this._signUpConfig.signUpFields) {
                    this.signUpFields = this._signUpConfig.signUpFields;
                }
                if (this._signUpConfig.header) {
                    this.header = this._signUpConfig.header;
                }
                if (this._signUpConfig.hiddenDefaults) {
                    this.hiddenFields = this._signUpConfig.hiddenDefaults;
                }
            }
        },
        enumerable: true,
        configurable: true
    });
    SignUpComponentCore.prototype.ngOnInit = function () {
        this.sortFields();
    };
    SignUpComponentCore.prototype.onSignUp = function () {
        var _this = this;
        // validate  required inputs
        var validation = this.validate();
        if (validation && validation.length > 0) {
            return this._setError("The following fields need to be filled out: " + validation.join(', '));
        }
        // create user attribute property
        this.user.attributes = {};
        // format phone number if it is a signUpField
        var phoneNumberRequested = this.signUpFields.find(function (el) {
            return el.key === 'phone_number';
        });
        if (phoneNumberRequested) {
            this.user.phone_number = "+" + this.country_code + this.local_phone_number;
        }
        // create user key and value arrays
        var userKeys = Object.keys(this.user);
        var userValues = userKeys.map(function (key) { return _this.user[key]; });
        // format data for Cognito user pool
        userKeys.forEach(function (key, index) {
            if (key !== 'username' && key !== 'password' && key !== 'attributes') {
                // needsPrefix determines if a custom attribute is indicated
                // and prepends 'custom:' to the key before sending to Cognito if needed
                var newKey = "" + (_this.needPrefix(key) ? 'custom:' : '') + key;
                _this.user.attributes[newKey] = userValues[index];
            }
        });
        this.amplifyService.auth()
            .signUp(this.user)
            .then(function (user) {
            var username = _this.user.username;
            _this.user = {};
            _this.amplifyService
                .setAuthState({ state: 'confirmSignUp', user: { 'username': username } });
        })
            .catch(function (err) { return _this._setError(err); });
    };
    SignUpComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    SignUpComponentCore.prototype.needPrefix = function (key) {
        var field = this.signUpFields.find(function (e) { return e.key === key; });
        if (key.indexOf('custom:') !== 0) {
            return field.custom;
        }
        else if (key.indexOf('custom:') === 0 && field.custom === false) {
            this.amplifyService.logger('SignUpComponent', 'WARN')
                .log('Custom prefix prepended to key but custom field flag is set to false');
        }
        return null;
    };
    SignUpComponentCore.prototype.onConfirmSignUp = function () {
        this.amplifyService
            .setAuthState({ state: 'confirmSignUp', user: { 'username': this.user.username } });
    };
    SignUpComponentCore.prototype.sortFields = function () {
        var _this = this;
        if (this.hiddenFields.length > 0) {
            this.defaultSignUpFields = this.defaultSignUpFields.filter(function (d) {
                return !_this.hiddenFields.includes(d.key);
            });
        }
        if (this._signUpConfig &&
            this._signUpConfig.signUpFields &&
            this._signUpConfig.signUpFields.length > 0) {
            if (!this._signUpConfig.hideAllDefaults) {
                // see if fields passed to component should override defaults
                this.defaultSignUpFields.forEach(function (f, i) {
                    var matchKey = _this.signUpFields.findIndex(function (d) {
                        return d.key === f.key;
                    });
                    if (matchKey === -1) {
                        _this.signUpFields.push(f);
                    }
                });
            }
            /*
                    sort fields based on following rules:
                    1. Fields with displayOrder are sorted before those without displayOrder
                    2. Fields with conflicting displayOrder are sorted alphabetically by key
                    3. Fields without displayOrder are sorted alphabetically by key
                  */
            this.signUpFields.sort(function (a, b) {
                if (a.displayOrder && b.displayOrder) {
                    if (a.displayOrder < b.displayOrder) {
                        return -1;
                    }
                    else if (a.displayOrder > b.displayOrder) {
                        return 1;
                    }
                    else {
                        if (a.key < b.key) {
                            return -1;
                        }
                        else {
                            return 1;
                        }
                    }
                }
                else if (!a.displayOrder && b.displayOrder) {
                    return 1;
                }
                else if (a.displayOrder && !b.displayOrder) {
                    return -1;
                }
                else if (!a.displayOrder && !b.displayOrder) {
                    if (a.key < b.key) {
                        return -1;
                    }
                    else {
                        return 1;
                    }
                }
            });
            this.signUpFields = this.removeHiddenFields();
        }
    };
    SignUpComponentCore.prototype.onAlertClose = function () {
        this._setError(null);
    };
    SignUpComponentCore.prototype.removeHiddenFields = function () {
        return this.signUpFields.filter(function (f) {
            return !f.displayOrder || f.displayOrder !== -1;
        });
    };
    SignUpComponentCore.prototype.validate = function () {
        var _this = this;
        var invalids = [];
        this.signUpFields.map(function (el) {
            if (el.key !== 'phone_number') {
                if (el.required && !_this.user[el.key]) {
                    el.invalid = true;
                    invalids.push(el.label);
                }
                else {
                    el.invalid = false;
                }
            }
            else {
                if (el.required && (!_this.country_code || !_this.local_phone_number)) {
                    el.invalid = true;
                    invalids.push(el.label);
                }
                else {
                    el.invalid = false;
                }
            }
        });
        return invalids;
    };
    SignUpComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    SignUpComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-sign-up-core',
                    template: template,
                },] },
    ];
    /** @nocollapse */
    SignUpComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
    SignUpComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
        "signUpConfig": [{ type: Input },],
    };
    return SignUpComponentCore;
}());
export { SignUpComponentCore };
