// aws-amplify-angular/dist/src/components/authenticator/forgot-password-component/forgot-password.component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-container\" *ngIf=\"_show\">\n  <div class=\"amplify-form-container\">\n    <div class=\"amplify-form-body\">\n    <div class=\"amplify-form-header\">Reset your password</div>\n    <div class=\"amplify-form-text\" *ngIf=\"!code_sent\">You will receive a verification code</div>\n    <div class=\"amplify-form-text\" *ngIf=\"code_sent\">Enter the code you received and set a new password</div>\n\n      <div class=\"amplify-form-row\" *ngIf=\"!code_sent\">\n        <input\n          (keyup)=\"setUsername($event.target.value)\"\n          class=\"amplify-form-input\"\n          type=\"text\"\n          placeholder=\"Username\"\n          [value]=\"username\"\n        />\n      </div>\n      <div class=\"amplify-form-row\" *ngIf=\"code_sent\">\n        <input #code\n          (keyup)=\"setCode(code.value)\"\n          class=\"amplify-form-input\"\n          type=\"text\"\n          placeholder=\"Enter code\"\n        />\n      </div>\n      <div class=\"amplify-form-row\" *ngIf=\"code_sent\">\n        <input #password\n          (keyup)=\"setPassword(password.value)\"\n          (keyup.enter)=\"onSubmit()\"\n          class=\"amplify-form-input\"\n          type=\"password\"\n          placeholder=\"Password\"\n        />\n      </div>\n\n      <div class=\"amplify-form-actions\">\n\n        <div class=\"amplify-form-cell-right\">\n          <button class=\"amplify-form-button\"\n            *ngIf=\"!code_sent\"\n            (click)=\"onSend()\">Submit</button>\n        \n          <button class=\"amplify-form-button\"\n            *ngIf=\"code_sent\"\n            (click)=\"onSubmit()\">Verify</button>\n        </div>\n\n        <div class=\"amplify-form-cell-left\">\n          <div class=\"amplify-form-actions-left\">\n            <a *ngIf=\"code_sent\" class=\"amplify-form-link\" (click)=\"onSend()\">Resend Code</a>\n            <a *ngIf=\"!code_sent\" class=\"amplify-form-link\" (click)=\"onSignIn()\">Back to Sign in</a>\n          </div>\n        </div>\n      \n      </div>\n    </div>\n  </div>\n\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ errorMessage }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n\n</div>\n";
var ForgotPasswordComponentCore = /** @class */ (function () {
    function ForgotPasswordComponentCore(amplifyService) {
        this.code_sent = false;
        this.amplifyService = amplifyService;
    }
    Object.defineProperty(ForgotPasswordComponentCore.prototype, "data", {
        set: function (data) {
            this._authState = data.authState;
            this._show = data.authState.state === 'forgotPassword';
            this.username = data.authState.user ? data.authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ForgotPasswordComponentCore.prototype, "authState", {
        set: function (authState) {
            this._authState = authState;
            this._show = authState.state === 'forgotPassword';
            this.username = authState.user ? authState.user.username || '' : '';
        },
        enumerable: true,
        configurable: true
    });
    ForgotPasswordComponentCore.prototype.setUsername = function (username) {
        this.username = username;
    };
    ForgotPasswordComponentCore.prototype.setCode = function (code) {
        this.code = code;
    };
    ForgotPasswordComponentCore.prototype.setPassword = function (password) {
        this.password = password;
    };
    ForgotPasswordComponentCore.prototype.onSend = function () {
        var _this = this;
        if (!this.username) {
            this.errorMessage = "Username cannot be empty";
            return;
        }
        this.amplifyService.auth().forgotPassword(this.username)
            .then(function () {
            _this.code_sent = true;
        })
            .catch(function (err) {
            _this._setError(err);
            _this.code_sent = false;
        });
    };
    ForgotPasswordComponentCore.prototype.onSubmit = function () {
        var _this = this;
        this.amplifyService.auth()
            .forgotPasswordSubmit(this.username, this.code, this.password)
            .then(function () {
            var user = { username: _this.username };
            _this.amplifyService.setAuthState({ state: 'signIn', user: user });
        })
            .catch(function (err) { return _this._setError(err); });
    };
    ForgotPasswordComponentCore.prototype.onSignIn = function () {
        this.amplifyService.setAuthState({ state: 'signIn', user: null });
    };
    ForgotPasswordComponentCore.prototype.onAlertClose = function () {
        this._setError(null);
    };
    ForgotPasswordComponentCore.prototype._setError = function (err) {
        if (!err) {
            this.errorMessage = null;
            return;
        }
        this.errorMessage = err.message || err;
    };
    ForgotPasswordComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-forgot-password-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    ForgotPasswordComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    ForgotPasswordComponentCore.propDecorators = {
        "data": [{ type: Input },],
        "authState": [{ type: Input },],
    };
    return ForgotPasswordComponentCore;
}());
export { ForgotPasswordComponentCore };
