// aws-amplify-angular/dist/src/components/authenticator/greeting-component/greeting.component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
var template = "\n<div class=\"amplify-greeting\" *ngIf=\"signedIn\">\n    <div class=\"amplify-greeting-text\">{{ greeting }}</div>\n    <div class=\"amplify-greeting-flex-spacer\"></div>\n    <a class=\"amplify-form-link amplify-greeting-sign-out\"\n      (click)=\"onSignOut()\"\n    >Sign out</a>\n</div>\n";
var GreetingComponentCore = /** @class */ (function () {
    function GreetingComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
        this.subscribe();
    }
    GreetingComponentCore.prototype.subscribe = function () {
        var _this = this;
        this.amplifyService.authStateChange$
            .subscribe(function (state) { return _this.setAuthState(state); });
    };
    GreetingComponentCore.prototype.setAuthState = function (authState) {
        this.authState = authState;
        this.signedIn = authState.state === 'signedIn';
        this.greeting = this.signedIn
            ? "Hello, " + authState.user.username
            : "";
    };
    GreetingComponentCore.prototype.onSignOut = function () {
        this.amplifyService.auth().signOut();
    };
    GreetingComponentCore.decorators = [
        { type: Component, args: [{
                    selector: 'amplify-auth-greetings-core',
                    template: template
                },] },
    ];
    /** @nocollapse */
    GreetingComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    GreetingComponentCore.propDecorators = {
        "authState": [{ type: Input },],
    };
    return GreetingComponentCore;
}());
export { GreetingComponentCore };
